src/datasources/asset/components/AssetQueryEditor.tsx

```diff
@@ -56,14 +58,22 @@ export function AssetQueryEditor({ query, onChange, onRunQuery, datasource }: Pr
       });
     }
     if (queryType === AssetQueryType.AssetSummary) {
-      handleQueryChange({ ...queryRef.current, queryType: AssetQueryType.AssetSummary, ...defaultAssetSummaryQuery });
+      handleQueryChange(
+        { ...queryRef.current, queryType: AssetQueryType.AssetSummary, ...defaultAssetSummaryQuery },
+        true
+      );
     }
   }, [queryType, handleQueryChange]);
 
   return (
     <div style={{ position: 'relative' }}>
       <InlineField label="Query type" labelWidth={22} tooltip={tooltips.queryType}>
-        <Select options={queryTypeOptions} onChange={handleQueryTypeChange} value={queryType} width={85} />
+        <Select
+          options={queryTypeOptions}
+          onChange={handleQueryTypeChange}
+          value={queryTypeOptions.find(option => option.value === queryType)}
```

Comment:
Was this change necessary ?

Reply:
Yes, due to the `setState` change which now points to a string and not AssetQueryType